zcash_primitives/src/note_encryption.rs

```diff
@@ -337,31 +395,51 @@ impl SaplingNoteEncryption {
 
 fn parse_note_plaintext_without_memo(
     ivk: &Fs,
+    epk: &edwards::Point<Bls12, PrimeOrder>,
```

Comment:
pass in optional `esk`, to be used when called by sender (i.e. when using outgoing viewing key)